authoring-recipes/recipe-conventions-and-best-practices.md

```diff
@@ -4,82 +4,153 @@ description: Conventions to follow, pitfalls to avoid
+
+## Best Practices
 
 ### Do No Harm
 
-If your Recipe cannot determine that a change is safe to make, such as when a type you're looking for is unavailable, it is always preferable to make no change than to make the wrong change. Relatedly, when making changes always strive to make the most minimal, least invasive version of that change. Keeping changes minimal makes for easy-to-review diffs and higher rates of pull request acceptance. A change that unnecessarily clobbers formatting or is otherwise overly broad will burden code reviewers and drastically reduce the rate at which they accept changes.
+One of the most important conventions to keep in mind when creating recipes is the concept of "Do No Harm". If your recipe cannot determine that a change is safe to me, **it should make no changes rather than making a potentially wrong change**. 
+
+For example, in the [RenameLocalVariableToCamelCase recipe](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/cleanup/RenameLocalVariablesToCamelCase.java), we prevent the recipe from renaming variables to a name that _could_ be a namespace conflict. Likewise, we opt out of renaming class fields since they _might_ be in use by some downstream API. We favor making fewer changes over making wrong changes.
+
+Relatedly, when making changes, always strive to make the most minimal, least invasive version of that change. For instance, in the [NoGuavaImmutableSetOf recipe](https://github.com/openrewrite/rewrite-migrate-java/blob/main/src/main/java/org/openrewrite/java/migrate/guava/NoGuavaImmutableSetOf.java), we restrict the recipe from changing any code that does not use at least Java 9.
+
+By reducing a recipe's scope and minimizing the number of changes a recipe makes, you'll make it easier for others to review the changes. This, in turn, will result in more pull requests being accepted and more trust in your recipes.
+
+On the other hand, a change that unnecessarily clobbers formatting or is otherwise overly broad in scope will burden code reviewers and drastically reduce the rate at which they accept changes.
 
 {% hint style="success" %}
-RewriteTest helps you to verify that your recipe does not make unnecessary changes by running your recipe in a loop. If you see your change being made many times it is likely your visitor fails to avoid making unnecessary changes.
+[RewriteTest](/authoring-recipes/recipe-testing.md#rewritetest-interface) helps you to verify that your recipe does not make unnecessary changes by running your recipe in a loop. If you see your change being made many times it is likely your visitor fails to avoid making unnecessary changes. See the [cycle section](#stay-single-cycle) for more information.
```

Comment:
I would replace "running your recipe in a loop" to "running your recipe over multiple cycles"